kong/plugins/cors/handler.lua

```diff
@@ -50,8 +56,25 @@ local function configure_origin(ngx, conf)
 
   local req_origin = ngx.var.http_origin
   if req_origin then
+
+    local parsed_req_origin_obj = url.parse(req_origin)
+    local parsed_req_origin = parsed_req_origin_obj and parsed_req_origin_obj.host 
```

Comment:
[RFC 6454](https://tools.ietf.org/html/rfc6454#page-11) says on "Comparing Origins" that:

>  Two origins are "the same" if, and only if, they are identical.  In
   particular: 
> If the two origins are scheme/host/port triples, the two origins
      are the same if, and only if, they have identical schemes, hosts,
      and ports.

Also see:

>  Q: Why not just use the host?
>   A: Including the scheme in the origin tuple is essential for
   security.  If user agents did not include the scheme, there would be
   no isolation between http://example.com and https://example.com
   because the two have the same host.  However, without this isolation,
   an active network attacker could corrupt content retrieved from
   http://example.com and have that content instruct the user agent to
   compromise the confidentiality and integrity of content retrieved
   from https://example.com, bypassing the protections afforded by TLS
   [RFC5246].

The comparisons we are doing now do not take scheme and port into consideration (nor do we have tests asserting them). See section 3.2.1 of RFC 6454 for examples of resources with similar and different origins, we could use this list to write our tests.